contracts/Swaplace.sol

```diff
@@ -52,12 +54,16 @@ contract Swaplace is SwapFactory, ISwaplace, IERC165 {
   function acceptSwap(uint256 swapId) public returns (bool) {
     Swap memory swap = _swaps[swapId];
 
-    if (swap.allowed != address(0) && swap.allowed != msg.sender)
+    (address allowed , uint256 expiry) = parseData(swap.config);
+
+    if (allowed != address(0) && allowed != msg.sender)
       revert InvalidAddress(msg.sender);
 
-    if (swap.expiry < block.timestamp) revert InvalidExpiry(swap.expiry);
+    if (expiry < block.timestamp) revert InvalidExpiry(expiry);
+
+    expiry = 0;
```

Comment:
This line is unnecessary as the `0` can be directly inputted in line 66 in the expiry variable of packData